### PR TITLE
Upgrade agent to 10.19.0.6562-1

### LINKED ISSUE #161: Support for 4.4 Mongodb versions
I tried to deploy cluster on kind with version set to 4.4.0 and it stuck on agent checking health.
It would be cool to support newest 4.4 mongo too.

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

closes #161

### CODE REVIEW

File: pkg/controller/mongodb/replica_set_controller.go [hunk 1]
```diff
@@ -394,7 +397,7 @@ func buildAutomationConfig(mdb mdbv1.MongoDB, mdbVersionConfig automationconfig.
 // TODO: Remove this once the agent doesn't require any config: https://jira.mongodb.org/browse/CLOUDP-66024.
 func dummyToolsVersionConfig() automationconfig.ToolsVersion {
 	return automationconfig.ToolsVersion{
-		Version: "100.0.2",
+		Version: envvar.GetEnvOrDefault(mongodbToolsVersionEnv, "100.1.0"),
```
Comment (chatton) [inline pkg/controller/mongodb/replica_set_controller.go:400]: allow us to specify tools version if needed

File: test/e2e/e2eutil.go [hunk 2]
```diff
@@ -130,8 +130,8 @@ func NewTestMongoDB(name string) (mdbv1.MongoDB, mdbv1.MongoDBUser) {
 		Spec: mdbv1.MongoDBSpec{
 			Members:                     3,
 			Type:                        "ReplicaSet",
-			Version:                     "4.0.6",
-			FeatureCompatibilityVersion: "4.0",
```
Comment (chatton) [inline test/e2e/e2eutil.go:134]: default to 4.4 with our tests

File: scripts/dev/templates/Dockerfile.operator [hunk 3]
```diff
@@ -4,7 +4,7 @@
 # TODO: This build takes longer than it needs to and can still be optimized
 RUN go build -o build/_output/bin/mongodb-kubernetes-operator github.com/mongodb/mongodb-kubernetes-operator/cmd/manager
 
-ENV manifest_version=4.2
+ENV manifest_version=4.4
```
Comment (chatton) [inline scripts/dev/templates/Dockerfile.operator:7]: update version manifest so it includes all of the 4.4 mongodb versions

File: pkg/util/envvar/envvars.go [hunk 4]
```diff
@@ -26,3 +27,10 @@ func MergeWithOverride(existing, desired []corev1.EnvVar) []corev1.EnvVar {
 	})
 	return mergedEnv
 }
+
+func GetEnvOrDefault(envVar, defaultValue string) string {
+	if val := os.Getenv(envVar); val != "" {
+		return val
+	}
+	return defaultValue
```
Comment (bznein) [inline pkg/util/envvar/envvars.go:35]: [nit] Instead on relying on the value returned to be empty, why not using [LookupEnv](https://golang.org/pkg/os/#LookupEnv)?

**Review by bznein — COMMENTED** [review]
Looks great, one nit and one question!

File: test/e2e/feature_compatibility_version_upgrade/feature_compatibility_version_upgrade_test.go [hunk 5]
```diff
@@ -27,12 +27,19 @@ func TestFeatureCompatibilityVersionUpgrade(t *testing.T) {
...
 	_, err := setup.GeneratePasswordForUser(user, ctx)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	_, err = setup.GeneratePasswordForUser(user, ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
```
[the remark this answers is not in the document]
Comment (chatton) [reply test/e2e/feature_compatibility_version_upgrade/feature_compatibility_version_upgrade_test.go:42]: nope looks like it got duplicated accidentally! Will remove!

File: pkg/util/envvar/envvars.go (hunk 4 above)
Comment (chatton) [reply pkg/util/envvar/envvars.go:35]: good catch, will switch to this

**Comment by chatton** [comment]
@bznein ready for another look!

**Pull request merged**